scripts/request

```diff
@@ -0,0 +1,157 @@
+require 'uri'
+require 'net/http'
+require 'json'
+require 'commander/import'
+
+class Request
+  def self.call(*args)
+    new(*args).call
+  end
+
+  def initialize(method, url, payload = {})
+    @method = method
+    @url = url
+    @payload = payload
+  end
+
+  def call
+    response = Net::HTTP.new(uri.host, uri.port).request(request)
+
+    unless response.code.start_with?('2')
+      $stderr.puts
+      $stderr.puts 'Error:'
+      $stderr.puts "  #{response.body}"
+      exit 1
+    else
+      puts response.body
+      puts 'Success.'
+    end
+  end
+
+  private
+
+  attr_reader :method, :url, :payload
+
+  def uri
+    @uri ||= URI.parse(url)
+  end
+
+  def request
+    case method
+    when :post
+      req = Net::HTTP::Post.new(uri.path, 'Content-Type': 'application/json')
+      req.body = payload.to_json
+      req
+    when :put
+      req = Net::HTTP::Put.new(uri.path, 'Content-Type': 'application/json')
+      req.body = payload.to_json
+      req
+    when :get
+      req = Net::HTTP::Get.new(uri.path, 'Content-Type': 'application/json')
+    end
+  end
+end
+
+def slice_and_compact(hash, *keys)
+  hash.select { |key, value| keys.include?(key) && !value.nil? }
+end
+
+program :name, 'Todo'
+program :version, '0.0.1'
+program :description, 'Interact with the Event Sourcery Todo Example App.'
+
+never_trace!
+
+script_name = $0
+
+command :add do |c|
+  c.syntax = 'Todo add [options]'
+  c.summary = 'Add a Todo item via the Todo web API'
+  c.example 'Add a Todo', %Q{#{script_name} add -t "My task"}
+  c.example 'Add a more complex Todo', %Q{#{script_name} add -t "My task" -d "My task description" -D 2017-01-01 -s stakeholder@example.com}
+  c.option '-t TITLE', '--title TITLE', 'Title'
+  c.option '-d DESCRIPTION', '--description DESCRIPTION', 'Description'
+  c.option '-D DUE_DATE', '--due_date DUE_DATE', 'Due date'
+  c.option '-s STAKEHOLDER_EMAIL', '--stakeholder_email STAKEHOLDER_EMAIL', 'Stakeholder email'
+  c.action do |args, options|
+    todo_id = SecureRandom.uuid
+    payload = slice_and_compact(options.default, :title, :description, :due_date, :stakeholder_email)
+    puts "Adding todo [#{todo_id}]: #{payload}"
+    Request.call(:post, "http://localhost:3000/todo/#{todo_id}", payload)
+  end
+end
+
+command :amend do |c|
+  c.syntax = 'Todo amend [options]'
+  c.summary = 'Amend a Todo item via the Todo web API'
+  c.example 'Amend the title of a Todo', %Q{#{script_name} add -i 0b341422-c516-4ee4-8f3e-ef1992dfff32 -t "My task"}
+  c.option '-i ID', '--id ID', 'Existing Todo ID'
+  c.option '-t TITLE', '--title TITLE', 'Title'
+  c.option '-d DESCRIPTION', '--description DESCRIPTION', 'Description'
+  c.option '-D DUE_DATE', '--due_date DUE_DATE', 'Due date'
+  c.option '-s STAKEHOLDER_EMAIL', '--stakeholder_email STAKEHOLDER_EMAIL', 'Stakeholder email'
+  c.action do |args, options|
+    todo_id = options.id
+    unless todo_id
+      $stderr.puts "Error: you must specify a todo_id to amend"
```

Comment:
`todo_id` is called "Todo ID" in line 91. Maybe this message should say something like "Error: Todo ID is not specified"?